src/dynamics/tendencies.jl

```diff
@@ -214,58 +327,55 @@ of the logarithm of surface pressure ln(p_s) and D̄ the vertically averaged div
 3. D̄ is subtracted in spectral space.
 4. Set tendency of the l=m=0 mode to 0 for better mass conservation."""
 function surface_pressure_tendency!(
-    diagn::DiagnosticVariables,
-    S::SpectralTransform,
-)
+        diagn::DiagnosticVariables,
+        S::SpectralTransform,
+    )
     (; pres_tend, pres_tend_grid) = diagn.tendencies
     (; ∇lnp_x, ∇lnp_y, u_mean_grid, v_mean_grid, div_mean, scratch_memory) = diagn.dynamics
-    
+
     # in grid-point space the the (ū, v̄)⋅∇lnpₛ term (swap sign in spectral)
     # += to allow for forcing contributions already in pres_tend_grid
-    @. pres_tend_grid += u_mean_grid*∇lnp_x + v_mean_grid*∇lnp_y
+    @. pres_tend_grid += u_mean_grid * ∇lnp_x + v_mean_grid * ∇lnp_y
 
     ūv̄∇lnpₛ = diagn.dynamics.a_2D           # reuse 2D work array
     transform!(ūv̄∇lnpₛ, pres_tend_grid, scratch_memory, S)
-    
+
     # for semi-implicit div_mean is calc at time step i-1 in vertical_integration!
     @. pres_tend -= ūv̄∇lnpₛ + div_mean      # add the -div_mean term in spectral, swap sign
-    
-    pres_tend[1] = 0                        # for mass conservation
+
+    pres_tend[1:1] .= 0                        # for mass conservation
```

Comment:
We also do this in `implicit_correction` note to self to check whether both are needed